tests/rptest/tests/raft_slow_follower_test.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from collections import defaultdict
+import random
+import signal
+import threading
+import time
+
+import numpy
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from confluent_kafka import Producer
+from confluent_kafka.cimpl import KafkaError
+from ducktape.utils.util import wait_until
+
+
+class ThreadedProducer:
+    def __init__(self, topic, redpanda, logger):
+        self.thread = threading.Thread(target=lambda: self.produce_loop())
+        self.thread.daemon = True
+        self.stop_ev = threading.Event()
+        self.redpanda = redpanda
+        self.logger = logger
+        self.producer = None
+        self.topic = topic
+        self.latencies = []
+        self.inflight = threading.Semaphore(20)
+
+    def start(self):
+        self.thread.start()
+
+    def produce_loop(self):
+        self.producer = Producer({
+            "bootstrap.servers": self.redpanda.brokers(),
+        })
+
+        def delivery_clb(err: KafkaError, msg, start_time):
+            if err:
+                self.logger.warn(f"error sending message: {err}")
+
+            self.latencies.append(time.monotonic() - start_time)
+            self.inflight.release()
+
+        while not self.stop_ev.is_set():
+            self.inflight.acquire()
+
+            start = time.monotonic()
+            self.producer.produce(
+                topic=self.topic,
+                key="test-message-key",
+                value="test-message-value",
+                on_delivery=lambda err, msg: delivery_clb(err, msg, start))
+            self.producer.flush()
+
+    def wait_for_messages(self, messages, timeout_sec):
+        wait_until(lambda: len(self.latencies) >= messages,
+                   timeout_sec=timeout_sec,
+                   backoff_sec=1,
+                   err_msg=f"timed out waiting for {messages} messages")
+
+    def stop(self):
+        self.stop_ev.set()
+        self.inflight.release()
+        self.thread.join()
+
+
+class RaftSlowFollowerTest(RedpandaTest):
```

Comment:
Awesome to see a test for this.

Did you check that this test fails without your Redpanda changes?

Reply:
yes, it does fail